Add fromOID to KeyWrapAlgorithm

cipherboy opened the pull request:
This is a copy of the `getKeyWrapAlgorithmFromOID` method from PKI's
`com.netscape.cmsutil.crypto.CryptoUtil`. This introduces new OID members
with public scope to allow future code to use the OID values (especially
under the `org.mozilla.jss.netscape.security` package).

`Signed-off-by: Alexander Scheel <ascheel@redhat.com>`

cipherboy commented:
This allows a clean implementation of `WrappingParams` from `netscape.security.util`:

```diff
 diff org/mozilla/jss/netscape/security/util/WrappingParams.java ../pki/base/util/src/netscape/security/util/WrappingParams.java 
1c1
< package org.mozilla.jss.netscape.security.util;
---
> package netscape.security.util;
12a13,14
> import com.netscape.cmsutil.crypto.CryptoUtil;
> 
63d64
< 
69c70
<         } else if (encryptOID.equals(KeyWrapAlgorithm.DES_CBC_PAD_OID.toString())) {
---
>         } else if (encryptOID.equals(CryptoUtil.KW_DES_CBC_PAD.toString())) {
71c72
<         } else if (encryptOID.equals(KeyWrapAlgorithm.AES_CBC_PAD_OID.toString())) {
---
>         } else if (encryptOID.equals(CryptoUtil.KW_AES_CBC_PAD.toString())) {
133c134
<         KeyWrapAlgorithm kwAlg = KeyWrapAlgorithm.fromOID(wrapOID);
---
>         KeyWrapAlgorithm kwAlg = CryptoUtil.getKeyWrapAlgorithmFromOID(wrapOID);
```

(Left is the one in #101, right is the one currently in PKI).

cipherboy commented on org/mozilla/jss/crypto/KeyWrapAlgorithm.java:
```diff
@@ -120,4 +122,25 @@ public int getBlockSize() {
+            return AES_CBC_PAD;
+
+        if (oid.equals(DES_CBC_PAD_OID))
+            return DES3_CBC_PAD;
```
It looks like (according to the search functionality), that [`1.3.14.3.2.7 = desCBC`](https://www.alvestrand.no/objectid/1.3.14.3.2.7.html) might be the correct OID arc for single DES.

The only other candidate is [`1.3.36.3.1.1.2 = DESCBC`](https://www.alvestrand.no/objectid/1.3.36.3.1.1.2.html), but that appears to be a specialized OID arc. Candidates from the [search page](https://www.alvestrand.no/cgi-bin/hta/oidwordsearch?text=DES&prefix=).

jmagne replied:
Yeah, I"m pretty sure we shouldn't be using single DES there. I'll look at the code and see what you are talking about though.

cipherboy replied:
To clarify: my latest thought is that the constant is ill-named, making the reader believe it is single DES, but it is really triple DES. The branch that isn't reachable appears to be for single DES (again, isn't reachable), but has a wrap algorithm of 3DES, so I'm confused about that.

jmagne replied:
Yeah, I think we should talk to alee about this. I just cant remember if this is something that was on purpose or just a bug. Everything seems to work ok, in practice since then. I did assist a bit in this effort at the time.

cipherboy commented:
So at minimum, I'm leaning towards renaming `DES_CBC_PAD_OID` to `DES3_CBC_PAD_OID` or `DES_EDE3_CBC_PAD_OID`.

cipherboy commented:
OK, I'm going ahead and pushing this if it passes CI + PKI check.

I've gone ahead and renamed the constant (`DES_CBC_PAD_OID -> DES3_CBC_PAD_OID`) and introduced a real `DES_CBC_PAD_OID=1.3.14.3.2.7` based off of [this arc](https://www.alvestrand.no/objectid/1.3.14.3.2.7.html) since it is used for other OIDs in PKI.

Note that there is little chance of this OID being used improperly as it is in a new location and requires a constant rename as well (`DES_CBC_PAD -> DES_CBC_PAD_OID`).

cipherboy commented:
This looks fine so I'm going ahead and merging. I've also updated the later code to use `DES3_CBC_PAD_OID` as well.
